eng/tox/run_mypy.py

```diff
@@ -61,24 +62,25 @@
     except CalledProcessError as src_err:
         src_code_error = src_err
 
-    if not is_check_enabled(args.target_package, "type_check_samples", True):
-        logging.info(
-            f"Package {package_name} opts-out of mypy check on samples."
-        )
-    else:
-        sample_code = [
-            *commands,
-            "--check-untyped-defs",
-            "--follow-imports=silent",
-            os.path.join(args.target_package, "samples")
-        ]
-        try:
+    if in_ci():
+        if not is_check_enabled(args.target_package, "type_check_samples", True):
             logging.info(
-                f"Running mypy commands on sample code: {sample_code}"
+                f"Package {package_name} opts-out of mypy check on samples."
             )
-            check_call(sample_code)
-        except CalledProcessError as sample_err:
-            sample_code_error = sample_err
+        else:
```

Comment:
Need to restructure this. It improperly makes it so that the sample code is only mypy-ed in CI, which is not the intent of this PR.